doc/source/whatsnew/v1.0.0.rst

```diff
@@ -301,6 +301,7 @@ Other
 - Using :meth:`DataFrame.replace` with overlapping keys in a nested dictionary will no longer raise, now matching the behavior of a flat dictionary (:issue:`27660`)
 - :meth:`DataFrame.to_csv` and :meth:`Series.to_csv` now support dicts as ``compression`` argument with key ``'method'`` being the compression method and others as additional compression options when the compression method is ``'zip'``. (:issue:`26023`)
 - :meth:`Series.append` will no longer raise a ``TypeError`` when passed a tuple of ``Series`` (:issue:`28410`)
+- :meth:`Series.pct_change` will can handle duplicate axis (:issue:`28664`)
```

Comment:
I thought the issue was caused by the anchored freq, but I found that the issue isn't as simple as I thought. for example, 

```python3
>>> import pandas as pd
>>> from pandas.tseries.frequencies import to_offset
>>> s = pd.Series([1, 2, 3, 4, 5]*3, pd.date_range('2019-09', periods=15, freq='1D'))
>>> to_offset('2B').isAnchored()
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-21-21cc41afbb94> in <module>
----> 1 s.pct_change(freq='2B')

~/pandas-0xF4D3C0D3/pandas/core/generic.py in pct_change(self, periods, fill_method, limit, freq, **kwargs)
  10415 
  10416         rs = data.div(data.shift(periods=periods, freq=freq, axis=axis, **kwargs)) - 1
> 10417         rs = rs.reindex_like(data)
  10418         if freq is None:
  10419             mask = isna(com.values_from_object(data))

~/pandas-0xF4D3C0D3/pandas/core/generic.py in reindex_like(self, other, method, copy, limit, tolerance)
   3929         )
   3930 
-> 3931         return self.reindex(**d)
   3932 
   3933     def drop(

~/pandas-0xF4D3C0D3/pandas/core/series.py in reindex(self, index, **kwargs)
   4136     @Appender(generic.NDFrame.reindex.__doc__)
   4137     def reindex(self, index=None, **kwargs):
-> 4138         return super().reindex(index=index, **kwargs)
   4139 
   4140     def drop(

~/pandas-0xF4D3C0D3/pandas/core/generic.py in reindex(self, *args, **kwargs)
   4559         # perform the reindex on the axes
   4560         return self._reindex_axes(
-> 4561             axes, level, limit, tolerance, method, fill_value, copy
   4562         ).__finalize__(self)
   4563 

~/pandas-0xF4D3C0D3/pandas/core/generic.py in _reindex_axes(self, axes, level, limit, tolerance, method, fill_value, copy)
   4580                 fill_value=fill_value,
   4581                 copy=copy,
-> 4582                 allow_dups=False,
   4583             )
   4584 

~/pandas-0xF4D3C0D3/pandas/core/generic.py in _reindex_with_indexers(self, reindexers, fill_value, copy, allow_dups)
   4622                 fill_value=fill_value,
   4623                 allow_dups=allow_dups,
-> 4624                 copy=copy,
   4625             )
   4626 

~/pandas-0xF4D3C0D3/pandas/core/internals/managers.py in reindex_indexer(self, new_axis, indexer, axis, fill_value, allow_dups, copy)
   1247         # some axes don't allow reindexing with dups
   1248         if not allow_dups:
-> 1249             self.axes[axis]._can_reindex(indexer)
   1250 
   1251         if axis >= self.ndim:

~/pandas-0xF4D3C0D3/pandas/core/indexes/base.py in _can_reindex(self, indexer)
   3282         # trying to reindex on an axis with duplicates
   3283         if not self.is_unique and len(indexer):
-> 3284             raise ValueError("cannot reindex from a duplicate axis")
   3285 
   3286     def reindex(self, target, method=None, level=None, limit=None, tolerance=None):

ValueError: cannot reindex from a duplicate axis
>>> # However the freq '2B' isn't anchored
>>> to_offset('2B').isAnchored()
False
>>> # Besides that, the freq of series itself also related to this issue.
>>> s = pd.Series([1, 2, 3, 4, 5]*3, pd.date_range('2019-09', periods=15, freq='1B'))
>>> s.pct_change(freq='2B')
2019-09-02         NaN
2019-09-03         NaN
2019-09-04    2.000000
2019-09-05    1.000000
2019-09-06    0.666667
2019-09-09   -0.750000
2019-09-10   -0.600000
2019-09-11    2.000000
2019-09-12    1.000000
2019-09-13    0.666667
2019-09-16   -0.750000
2019-09-17   -0.600000
2019-09-18    2.000000
2019-09-19    1.000000
2019-09-20    0.666667
Freq: B, dtype: float64
```

So I take an approach that dropping duplicate indices instead of handling anchored freq.